sd/etcd/registrar.go

```diff
@@ -43,6 +67,22 @@ func (r *Registrar) Register() {
 	} else {
 		r.logger.Log("action", "register")
 	}
+	if r.service.TTL == nil {
+		return
+	}
+	r.quit = make(chan struct{})
```

Comment:
This mutates registrar state, which makes it unsafe for concurrent access from multiple goroutines. I think the registrar needs its internal state protected with a mutex, which should be taken with every public method.